db/generateData.js

```diff
@@ -32,11 +31,11 @@ var writeData = (nrecords) => {
       var randInd = Math.floor(Math.random() * imgUrls.length);
       var randUrl = imgUrls[randInd];
       if (listingId === nrecords) {
-        writer.write([globalId, imageId, listingId, randUrl]);
+        writer.write([listingId, imageId, randUrl]);
```

Comment:
Did you not have to use the drain event?

Reply:
it's in there at the bottom